packages/wallet/test/services/ProviderTracker/HandleProvider.test.ts

```diff
@@ -0,0 +1,105 @@
+/* eslint-disable no-magic-numbers */
+/* eslint-disable camelcase */
+import { KoraLabsHandleProvider } from '../../../src/services/ProviderTracker';
+import { createGenericMockServer, mockProviders as mocks } from '@cardano-sdk/util-dev';
+import url from 'url';
+
+const bobHandle = {
+  hasDatum: false,
+  issuer: 'KoraLabs',
+  name: 'bob',
+  resolved_addresses: {
+    ada: 'addr_test1qzrljm7nskakjydxlr450ktsj08zuw6aktvgfkmmyw9semrkrezryq3ydtmkg0e7e2jvzg443h0ffzfwd09wpcxy2fuql9tk0g'
+  },
+  resolvedAt: 'zrljm7nskakjydxlr450ktsj08zuw6aktvgfkmm'
+};
+
+const aliceHandle = {
+  hasDatum: false,
+  issuer: 'KoraLabs',
+  name: 'alice',
+  resolved_addresses: {
+    ada: 'addr_test1qqk4sr4f7vtqzd2w90d5nfu3n59jhhpawyphnek2y7er02nkrezryq3ydtmkg0e7e2jvzg443h0ffzfwd09wpcxy2fuqmcnecd'
+  },
+  resolvedAt: 'zrljm7nskakjydxlr450ktsj08zuw6aktvgssmm'
+};
+
+export const mockServer = createGenericMockServer((handler) => async (req, res) => {
+  const result = await handler(req);
+
+  res.setHeader('Content-Type', 'application/json');
+  res.setHeader('Access-Control-Allow-Origin', '*');
+
+  if (result.body) {
+    res.statusCode = result.code || 200;
+    return res.end(JSON.stringify(result.body));
+  }
+
+  const reqUrl = url.parse(req.url!).pathname;
+
+  switch (reqUrl) {
+    case '/handles/bob':
+    case '/handles/alice': {
+      return res.end(JSON.stringify(result.body));
+    }
+    // No default
+  }
+
+  return res.end(JSON.stringify(result.body));
+});
+
+describe('HandleProvider', () => {
+  it('should resolve a handle', async () => {
+    const { serverUrl, closeMock } = await mockServer(async () => ({
+      body: bobHandle
+    }));
+    const provider = new KoraLabsHandleProvider({
+      networkInfoProvider: mocks.mockNetworkInfoProvider2(100),
+      serverUrl
+    });
+    const tip = await mocks.mockNetworkInfoProvider2(100).ledgerTip();
+    const result = await provider.resolveHandles({ handles: ['bob'] });
```

Comment:
This is a good indication of the existing test style we're using:
https://github.com/input-output-hk/cardano-js-sdk/blob/6e28df412797974b8ce6f6deb0c3346ff5938a05/packages/cardano-services-client/test/AssetInfoProvider/assetInfoHttpProvider.test.ts#L49-L53